src/cli/args.rs

```diff
@@ -135,11 +136,12 @@ impl<C: Clone + Eq + Debug + Subcommand, O: DescriptorOpts> Args<C, O> {
             };
 
         if sync {
-            let indexer = match (&self.resolver.esplora, &self.resolver.electrum) {
-                (None, Some(url)) => AnyIndexer::Electrum(Box::new(electrum::Client::new(url)?)),
-                (Some(url), None) => {
-                    AnyIndexer::Esplora(Box::new(esplora::Builder::new(url).build_blocking()?))
+            let indexer = match (&self.resolver.esplora, &self.resolver.electrum, &self.resolver.mempool) {
+                (None, Some(url), None) => AnyIndexer::Electrum(Box::new(electrum::Client::new(url)?)),
+                (Some(url), None, None) => {
+                    AnyIndexer::Esplora(Box::new(IndexerClient::new_esplora(url)?))
                 }
+                (None, None, Some(url)) => AnyIndexer::Mempool(Box::new(IndexerClient::new_mempool(url)?)),
                 _ => {
                     eprintln!(
                         " - error: no blockchain indexer specified; use either --esplora or \
```

Comment:
```suggestion
                        " - error: no blockchain indexer specified; use either --esplora, --mempool or \
```

Reply:
I've handled this command line prompt in my latest commit